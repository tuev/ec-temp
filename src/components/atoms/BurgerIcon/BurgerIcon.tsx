import React, { FC } from 'react'
import { AppBurgerIcon } from './BurgerIcon.styled'
import { BaseBurgerIconProps } from './BurgerIcon.types'

const BurgerIcon: FC<BaseBurgerIconProps> = (props: BaseBurgerIconProps) => (
  <AppBurgerIcon {...props}>
    <div />
    <div />
    <div />
  </AppBurgerIcon>
)

export default BurgerIcon
