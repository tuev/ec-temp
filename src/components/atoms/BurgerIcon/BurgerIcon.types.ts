import { SpaceProps } from 'styled-system'
import { ColorType } from 'theme/colors'

export interface IBurgerIcon {
  open?: boolean
  size?: number | string
  color?: ColorType
}

export declare type BaseBurgerIconProps = IBurgerIcon &
  SpaceProps &
  React.HTMLAttributes<HTMLButtonElement>
