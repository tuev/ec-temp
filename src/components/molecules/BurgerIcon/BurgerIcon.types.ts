import { SpaceProps } from 'styled-system'
import { ColorType } from 'theme/colors'

export interface IBurgerIcon {
  open?: boolean
  size?: number | string
  color?: ColorType
  handleOpen?: (openStatus: boolean) => void
}

export declare type BaseBurgerIconProps = IBurgerIcon &
  SpaceProps &
  React.HTMLAttributes<HTMLButtonElement>
