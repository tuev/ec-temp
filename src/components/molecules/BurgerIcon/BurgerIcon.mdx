import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './BurgerIcon.stories.tsx'
import BurgerIcon from './BurgerIcon.tsx'

<Story name="basic" />
<Story name="opened" />
<Story name="demo" />
