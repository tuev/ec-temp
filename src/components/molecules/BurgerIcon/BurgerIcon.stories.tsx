import React, { useState, useCallback } from 'react'
import { action } from '@storybook/addon-actions'
import BurgerIcon from './BurgerIcon'

export default {
  component: BurgerIcon,
  includeStories: ['basic', 'opened', 'demo'],
  title: 'BurgerIcon',
}

const BurgerIconDemo = () => {
  const [open, toggleOpen] = useState(false)

  const handleClick = useCallback(() => toggleOpen(!open), [open])
  return <BurgerIcon onClick={handleClick} open={open} />
}

export const basic = () => <BurgerIcon onClick={action('Button Clicked!!!')} />
export const opened = () => (
  <BurgerIcon open={true} onClick={action('Button Clicked!!!')} />
)
export const demo = () => <BurgerIconDemo />
