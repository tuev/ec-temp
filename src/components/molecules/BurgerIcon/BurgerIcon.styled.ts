import { prop } from 'styled-tools'
import styled from 'styled-components'

import { space } from 'styled-system'
import { BaseBurgerIconProps } from './BurgerIcon.types'
import { COLOR, ColorValues } from 'theme/colors'

export const AppBurgerIcon: React.FC<BaseBurgerIconProps> = styled.button<
  BaseBurgerIconProps
>`
  ${space};
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: ${prop('size', '2rem')};
  height: ${prop('size', '2rem')};
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 10;

  &:focus {
    outline: none;
  }

  div {
    width: ${prop('size', '2rem')};
    height: 0.25rem;
    background: ${({ color = 'primary' }): ColorValues => COLOR[color]};
    border-radius: 10px;
    transition: all 0.3s linear;
    position: relative;
    transform-origin: 1px;

    &:first-child {
      transform: ${({ open }): string =>
        open ? 'rotate(45deg)' : 'rotate(0)'};
    }

    &:nth-child(2) {
      opacity: ${({ open }): string => (open ? '0' : '1')};
      transform: ${({ open }): string =>
        open ? 'translateX(20px)' : 'translateX(0)'};
    }

    &:nth-child(3) {
      transform: ${({ open }): string =>
        open ? 'rotate(-45deg)' : 'rotate(0)'};
    }
  }
`
