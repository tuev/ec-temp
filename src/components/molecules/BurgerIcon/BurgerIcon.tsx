import React, { FC, useState, useCallback, useEffect } from 'react'
import { AppBurgerIcon } from './BurgerIcon.styled'
import { BaseBurgerIconProps } from './BurgerIcon.types'

const BurgerIcon: FC<BaseBurgerIconProps> = (props: BaseBurgerIconProps) => {
  const { open = false, handleOpen } = props
  const [isOpen, toggleOpen] = useState<boolean>(open)

  const handleClick = useCallback(() => toggleOpen(!isOpen), [isOpen])

  useEffect(() => {
    if (handleOpen) {
      handleOpen(isOpen)
    }
  }, [handleOpen, isOpen])

  useEffect(() => {
    // if (open !== isOpen) {
    toggleOpen(open)
    // }
  }, [open])

  return (
    <AppBurgerIcon
      {...props}
      open={isOpen}
      onClick={handleClick}
      data-testid="burger-icon"
    >
      <div />
      <div />
      <div />
    </AppBurgerIcon>
  )
}

export default BurgerIcon
