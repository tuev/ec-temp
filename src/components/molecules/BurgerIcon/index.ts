import BurgerIcon from './BurgerIcon'

export { BurgerIcon }

export default BurgerIcon
