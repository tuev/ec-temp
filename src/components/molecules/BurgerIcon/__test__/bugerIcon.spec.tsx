import React from 'react'
import BurgerIcon from '..'
import { mount } from 'enzyme'
import { render, fireEvent } from '@testing-library/react'

// NOTE:: Using mount from enzyme to test atoms that we can test any css changed in children dom
// Ref: https://blog.halolabs.io/testing-react-components-with-jest-enzyme-5d1dd4ddccc4
describe('Burger icon', () => {
  it('basic burger icon action', () => {
    const initValue = false
    let burgerStatus = initValue
    const handleClickBurger = jest
      .fn()
      .mockImplementation(value => (burgerStatus = value))
    const basicBurgerIcon = render(
      <BurgerIcon handleOpen={handleClickBurger} />
    )
    expect(basicBurgerIcon).toMatchSnapshot()

    const burgerBtn = basicBurgerIcon.getByTestId('burger-icon')
    fireEvent.click(burgerBtn)
    expect(handleClickBurger).toBeCalled()
    expect(burgerStatus).toEqual(!initValue)
  })

  it('basic burger icon', () => {
    const basicBurgerIcon = mount(<BurgerIcon />)
    expect(basicBurgerIcon).toMatchSnapshot()
  })
  it('opned burger icon btn', () => {
    const openedBurgerIcon = mount(<BurgerIcon open={true} />)
    expect(openedBurgerIcon).toMatchSnapshot()
  })
  it('green green', () => {
    const greenBurgerIcon = mount(<BurgerIcon color="green" />)
    expect(greenBurgerIcon).toMatchSnapshot()
  })
})
