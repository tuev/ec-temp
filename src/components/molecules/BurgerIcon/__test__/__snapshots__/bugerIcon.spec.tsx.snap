// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Burger icon basic burger icon 1`] = `
.c0 {
  position: absolute;
  top: 5%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 10;
}

.c0:focus {
  outline: none;
}

.c0 div {
  width: 2rem;
  height: 0.25rem;
  background: #F62F5E;
  border-radius: 10px;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  position: relative;
  -webkit-transform-origin: 1px;
  -ms-transform-origin: 1px;
  transform-origin: 1px;
}

.c0 div:first-child {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c0 div:nth-child(2) {
  opacity: 1;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.c0 div:nth-child(3) {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

<BurgerIcon>
  <styled.button
    data-testid="burger-icon"
    onClick={[Function]}
    open={false}
  >
    <button
      className="c0"
      data-testid="burger-icon"
      onClick={[Function]}
      open={false}
    >
      <div />
      <div />
      <div />
    </button>
  </styled.button>
</BurgerIcon>
`;

exports[`Burger icon basic burger icon action 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: absolute;
  top: 5%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 10;
}

.c0:focus {
  outline: none;
}

.c0 div {
  width: 2rem;
  height: 0.25rem;
  background: #F62F5E;
  border-radius: 10px;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  position: relative;
  -webkit-transform-origin: 1px;
  -ms-transform-origin: 1px;
  transform-origin: 1px;
}

.c0 div:first-child {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c0 div:nth-child(2) {
  opacity: 1;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.c0 div:nth-child(3) {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

<body>
    <div>
      <button
        class="c0"
        data-testid="burger-icon"
      >
        <div />
        <div />
        <div />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-AykKC edGmea"
      data-testid="burger-icon"
    >
      <div />
      <div />
      <div />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Burger icon green green 1`] = `
.c0 {
  position: absolute;
  top: 5%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 10;
}

.c0:focus {
  outline: none;
}

.c0 div {
  width: 2rem;
  height: 0.25rem;
  background: #00D3CA;
  border-radius: 10px;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  position: relative;
  -webkit-transform-origin: 1px;
  -ms-transform-origin: 1px;
  transform-origin: 1px;
}

.c0 div:first-child {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c0 div:nth-child(2) {
  opacity: 1;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.c0 div:nth-child(3) {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

<BurgerIcon
  color="green"
>
  <styled.button
    color="green"
    data-testid="burger-icon"
    onClick={[Function]}
    open={false}
  >
    <button
      className="c0"
      color="green"
      data-testid="burger-icon"
      onClick={[Function]}
      open={false}
    >
      <div />
      <div />
      <div />
    </button>
  </styled.button>
</BurgerIcon>
`;

exports[`Burger icon opned burger icon btn 1`] = `
.c0 {
  position: absolute;
  top: 5%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 10;
}

.c0:focus {
  outline: none;
}

.c0 div {
  width: 2rem;
  height: 0.25rem;
  background: #F62F5E;
  border-radius: 10px;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  position: relative;
  -webkit-transform-origin: 1px;
  -ms-transform-origin: 1px;
  transform-origin: 1px;
}

.c0 div:first-child {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c0 div:nth-child(2) {
  opacity: 0;
  -webkit-transform: translateX(20px);
  -ms-transform: translateX(20px);
  transform: translateX(20px);
}

.c0 div:nth-child(3) {
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

<BurgerIcon
  open={true}
>
  <styled.button
    data-testid="burger-icon"
    onClick={[Function]}
    open={true}
  >
    <button
      className="c0"
      data-testid="burger-icon"
      onClick={[Function]}
      open={true}
    >
      <div />
      <div />
      <div />
    </button>
  </styled.button>
</BurgerIcon>
`;
